storage/remote/codec.go

```diff
@@ -289,6 +291,10 @@ func validateLabelsAndMetricName(ls labels.Labels) error {
 		if !model.LabelValue(l.Value).IsValid() {
 			return errors.Errorf("invalid label value: %v", l.Value)
 		}
+		if l.Name == prevLabelName {
```

Comment:
You could avoid having prevLabelName by getting the index from the for, checking that it's >0 and then looking at `ls[i-1]`

Reply:
Sure, I guess its cleaner that way.